components/admin_console/group_settings/group_details/group_details.jsx

```diff
@@ -115,14 +132,59 @@ export default class GroupDetails extends React.PureComponent {
         this.setState({loadingTeamsAndChannels: false});
     }
 
-    onToggle = async (allowReference) => {
-        this.setState({allowReference});
-        this.props.actions.patchGroup(this.props.groupID, {allow_reference: allowReference});
+    onMentionToggle = async (allowReference) => {
+        const {group} = this.props;
+        const isOriginalAllowReference = group.allow_reference;
+
+        this.setState({
+            saveNeeded: true,
+            allowReference,
+            hasAllowReferenceChanged: allowReference !== isOriginalAllowReference}
+        );
+    }
+
+    onMentionChange = async (e) => {
+        const {group} = this.props;
+        const isOriginalGroupMentionName = group.name;
+        const groupMentionName = e.target.value;
+        const saveNeeded = true;
+
+        this.setState({
+            saveNeeded,
+            groupMentionName,
+            hasGroupMentionNameChanged: groupMentionName !== isOriginalGroupMentionName
+        });
+        this.props.actions.setNavigationBlocked(saveNeeded);
+    }
+
+    handleSubmit = async () => {
+        this.setState({saving: true});
+        const {allowReference, groupMentionName, hasAllowReferenceChanged, hasGroupMentionNameChanged} = this.state;
+
+        let serverError = null;
+        let saveNeeded = false;
+
+        if (!groupMentionName || groupMentionName.length === 0) {
+            saveNeeded = true;
+            serverError = <NeedGroupNameError/>;
+            this.setState({allowReference, serverError, saving: false, saveNeeded});
+        } else if (hasAllowReferenceChanged || hasGroupMentionNameChanged) {
+            saveNeeded = false;
+            const result = await this.props.actions.patchGroup(this.props.groupID, {allow_reference: allowReference, name: groupMentionName});
+            if (result.error) {
+                saveNeeded = true;
+                if (result.error.server_error_id === 'store.sql_group.unique_constraint') {
```

Comment:
Ah I'm not 100% if we want to be introducing backend errors into frontend in such manner. I'm okay with it but I will let @mkraft make a comment. Not sure if there is a better way to handle it though so let's wait to see what others have to say about it.

Reply:
I saw this pattern used in some other places in the client, therefore I used it. The thing is, the server does not have the context that the groupname is used as a mention, it only knows that there is a duplicate name. The context of that error exists at the client.

Reply:
I guess if it's being used in other places then it's fine to be used here as well. I just thought since we had discussions in developer meetings regarding like backend errors bubbling up to front end and wanting to avoid this is why I thought we shouldn't be doing it. @mkraft any thoughts?

Reply:
It could be that that the discussion was about them not bubbling through unchecked - in this case we are converting it. to a client error. I wish there was a more streamlined way of doing it, however this is the pattern I found.

Reply:
True say. Okay in that case it's fine then.